### DISCUSSION TITLE
Use PxWeb to get data from Swedish Konjunkturinstitutet

### DISCUSSION THREAD

**Discussion by MrNoClue** [rOpenGov/pxweb#263] [Q&A] [answered]
Can you use PxWeb to get data from Konjunkturinstitutet?
Apparently it also uses what SCB use, PxAPI 1.0.
My understanding is that you can, but I don't know what input to give into the list or how the URL should look like.

The matrix-code, that is found before you make any selection is not as unique as in e.g. SCB, in SCB it can be (under ContentsCode): 

> pxweb_query_list <- 
  list("Region"=c("22"),
       "ContentsCode"=c("NV1401A3")) #MATRIS-kod

But in konjunkturinstitutet it seems more general: 

> pxweb_query_list <- 
  list("Region"=c("SE32"),
       "Fråga" =c("207"),
       "Serie" =c("S"),
       "ContentsCode"=c("barindreg")) #MATRIS-kod

It seems like ContetnsCode from konj, basically says nothing. Instead it is "Fråga" that points to a specific series? ContentsCode can thus be removed?

When I run this code I get an error:

> pxweb_query_list <- 
  list("Region"=c("SE32"),
       "Fråga" =c("207"),
       "Serie" =c("S"),
       "ContentsCode"=c("barindreg"))

> px_data <- pxweb_get(url = "http://statistik.konj.se/PxWeb/api/v1/sv/KonjBar/ftgregional/Barindreg",
            query = pxweb_query_list)

It says: "Error in pxweb_advanced_get(url = url, query = query, verbose = verbose) : 
  Not Found (HTTP 404)." - what is wrong?

Trying to download this series from: http://statistik.konj.se/PxWeb/pxweb/sv/KonjBar/KonjBar__ftgregional/Barindreg.px/table/tableViewLayout1/

Thanks in advance!

**Comment by MansMeg** [comment, marked as answer]
Have you tried the interactive tool to setup your query?
```
dat <- pxweb_interactive("statistik.konj.se")
```
I got this to work, but I don't know if that's the data you want. But it worked for me:
```
# PXWEB query 
pxweb_query_list <- 
  list("Region"=c("SE11","SE12"),
       "Fråga"=c("con","101"),
       "Serie"=c("O"),
       "Period"=c("2007Q1","2007Q2","2007Q3"))

# Download data 
px_data <- 
  pxweb_get(url = "http://statistik.konj.se/PXWeb/api/v1/sv/KonjBar/ftgregional/Barindreg.px",
            query = pxweb_query_list)
```

**Reply by MrNoClue** [reply]
Thanks! No I have not tried the interactive tool before, runned that code - but you can only exit?

Anyways with suggested changes from your code it worked:

- really thought you should left out the ".px" in the URL - apparently not
- when trying to send Contentscode into pxweb_get (w correct URL) is says you can't send that, so removing that from the pxweb_query_list I get the px_data I want

Thanks for the help! :)

**Reply by thelinkan** [reply]
There is a difference between SCB and Konjunkturinstitutet. SCB uses a SQL database in the backend, while Konjunkturinstitutet uses PX-files. When using files, you need to include .px in the URL.

**Reply by MrNoClue** [reply]
Thanks for clarification @thelinkan , apparently you should follow instructions - not do modifications when you not know what you're doing or why (!) :P
